Tabs: Try a simpler tab focus style, alt

jasmussen opened the pull request:
## What?

Alternative to #46030. Explores a simpler tab focus style that isn't as heavy as it currently is. 

Before:

<img src="https://user-images.githubusercontent.com/1204802/203772657-d2103fce-e231-405a-88b1-267d2e560f4c.png">

After:
![tabs](https://user-images.githubusercontent.com/1204802/205290192-d98499d2-f080-4912-8100-2772609ae1c4.gif)

## Testing Instructions + Testing Instructions for Keyboard

Navigate through tabs in the inspector and inserter and observe the simpler focus style. Be sure to test also the vertical orientation: https://wordpress.github.io/gutenberg/?path=/story/components-tabpanel--default&args=orientation:vertical — focus style should be the same.

jasmussen committed:
Initial focus style refactor.

jasmussen committed:
Try alternate style.

mirka commented on packages/components/src/tab-panel/style.scss:
```diff
@@ -7,57 +7,71 @@
-	// See: https://github.com/WordPress/gutenberg/pull/9793
+	&:focus:not(:disabled) {
+		position: relative;
+		z-index: z-index(".edit-post-sidebar__panel-tab.is-active");
```
I don't quite understand the purpose of all the z-index styles. Could you explain the intent?

jasmussen replied:
I'm actually not sure — I know it was necessary at one point, but I think the tabs have been refactored enough that it might not be needed anymore. When I tried removing it just now, it worked fine without it. I'll try and remove it and see if that helps.

mirka commented on packages/components/src/tab-panel/style.scss:
```diff
@@ -7,57 +7,71 @@
+		pointer-events: none;
+
+		// Draw the indicator.
+		background: var(--wp-admin-theme-color);
```
In the wp-components package, we've transitioned to this theme-ready variable:

```suggestion
		background: $components-color-accent;
```
(Maybe we need a lint rule for this... 🤔)

Anything outside of the wp-components package, we can continue to use `var(--wp-admin-theme-color)` for the time being 👍

mirka commented on packages/components/src/tab-panel/style.scss:
```diff
@@ -7,57 +7,71 @@
-	&.is-active:focus {
-		box-shadow: inset 0 0 0 var(--wp-admin-border-width-focus) $components-color-accent, inset 0 -#{$border-width-tab * 2} 0 0 $components-color-accent;
+	&:focus-visible::before {
+		box-shadow: 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color);
```
```suggestion
		box-shadow: 0 0 0 var(--wp-admin-border-width-focus) $components-color-accent;
```

mirka reviewed (commented):
This is testing well for me 💯 

I just want to clear up the z-index stuff, because there are some anti-patterns here that I'd like to address before merging if the z-index styles are really necessary.

And if you could add a brief [changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md) for the TabPanel component, that would be great.

jasmussen committed:
Update packages/components/src/tab-panel/style.scss

Co-authored-by: Lena Morita <lena@jaguchi.com>

jasmussen committed:
Update packages/components/src/tab-panel/style.scss

Co-authored-by: Lena Morita <lena@jaguchi.com>

jasmussen committed:
Remove z-index, update tabs in site editor as well.

jasmussen committed:
Update changelog.

jasmussen commented:
Thanks so much for the review, and for the suggestions. I've removed the z-index, and updated the changelog. I also noticed that the site editor had its own copy of the tab styles, so I updated them there as well.

One thing I noted, and could use your advice on, is that the separate navigation screen code is apparently still in the codebase, and uses the old styles. I don't think this code is meant to be used anymore, so I didn't update it. @Mamaduka do you have any insights on what to best do here? Without updating that code, we can't remove the z-index layer from base-styles, this one: `".edit-post-sidebar__panel-tab.is-active": 1,` — so it might be best to update the code after all? 🤔

Mamaduka commented:
Thanks for the ping, @jasmussen. Can you give me a link to the navigation screen code in question?

jasmussen commented:
Sorry I should've included that: https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-navigation/src/components/sidebar/style.scss#L57 

In fact most of the tab code there is again just a verbatim copy of tab code being updated in this PR. It'd be nice to find a way to do more code reuse in the future, so it doesn't drift.

Speaking of, I'll look into the widgets screen, I can't recall if it has tabs 🙈

jasmussen committed:
Update widgets screen tabs.

jasmussen commented:
I actually went ahead and updated both the widget and navigation screen tabs. It's unfortunate there's so much code duplication, but it doesn't seem like there's great value in holding on to the z-index layer if just obsolete code uses it. Let me know what you think and I can revert.

Mamaduka commented:
Thanks, Joen. I saw the latest commit, so I guess that resolves the `z-index` issue :)

The Navigation Screen is no longer maintained (#43620), and I believe there's no way to access it via UI. So it's probably okay to break styles there.

jasmussen commented:
Thanks for the sanity check.

I wasn't able to check, but the CSS I pushed shouldn't actually break anything, it's just that no-one will see it. But I had to make a change, because the code still gets compiled, and if it references a missing SCSS variable it breaks.

mirka commented on packages/edit-widgets/src/components/sidebar/style.scss:
```diff
@@ -21,59 +21,74 @@
 
+// This tab style CSS is duplicated verbatim in
+// /packages/components/src/tab-panel/style.scss
 .components-button.edit-widgets-sidebar__panel-tab {
```
Asking for future reference — Is there an easy way to test this widget screen? Do I need some kind of widget-enabled theme?

Mamaduka replied:
Any default theme before 2022 should have widgets enabled.

mirka replied:
Ah, got it. Thank you!

mirka reviewed (approved):
Looks good! 🚢 Thanks for the z-index clean up.

We'll be tackling the code duplication soon (#43414).

jasmussen commented:
I'll trust these reviews then, thank you!

afercia commented:
Nice improvement! I'd suggest to move also the transparent outline used to support Windows High Contrast Mode from the button to the pseudo element, otherwise it's cut-off because one of the ancestors uses `overflow: auto`. See the screenshot below, where I made the outline red for clarity.

Worth reminding in Windows High Contrast Mode the CSS box-shadows aren't visible. The outline is. 

<img width="309" alt="Screenshot 2022-12-14 at 09 02 38" src="https://user-images.githubusercontent.com/1682452/207541326-974ab38c-ec6d-46fa-9e04-4d0eefa45556.png">

jasmussen commented:
I'll do that in a followup when I get a moment.

mtias commented:
Thanks for working on this one!
